security/cmd/istio_ca/main.go

```diff
@@ -70,9 +70,6 @@ type cliOptions struct { // nolint: maligned
 	workloadCertGracePeriodRatio float32
 	// The minimum grace period for workload cert rotation.
 	workloadCertMinGracePeriod time.Duration
-
-	// TODO(incfly): delete this field once we deprecate flag --grpc-hostname.
-	grpcHostname string
```

Comment:
I would prefer if instead of removing this flag - we would implement it. 
Right now citadel is exposed with a hardcoded and atypical name - which creates various problems. It is better to allow users to customize it to match the real domain exposed by mesh expansion.

It is a pretty bad practice to not follow the TLS standards and common practices - having DNS domains with one name presenting a cert with a different name, and overriding it in client is not very elegant.